dealership/vehicles.py

```diff
@@ -1,15 +1,36 @@
 class Vehicle(object):
     def __init__(self, maker, model, year, base_price, miles):
-        pass
+        self.maker = maker
+        self.model = model
+        self.year = year
+        self.base_price = base_price
+        self.miles = miles
 
 
 class Car(Vehicle):
-    pass
-
+    def sale_price(self):
+        return self.base_price * 1.2
+    
+    def purchase_price(self):
+        sale_price = self.sale_price()
+        return sale_price - (0.004 * self.miles)
+        
 
 class Motorcycle(Vehicle):
-    pass
-
+    def sale_price(self):
+        return self.base_price * 1.1
+    
```

Comment:
You are using the the <span style="color:grey">sale_price()</span> and <span style="color:grey">purchase_price() </span>methods multiple times for each vehicle. You could implement them within the Vehicle class instead, but you'd need to declare the multipliers as constants within the Car, Motorcycle and Truck classes.